pkg/provider/ip/eni/eni.go

```diff
@@ -169,8 +169,11 @@ func (e *eniManager) CreateIPV4Resource(required int, resourceType config.Resour
 
 	// If the existing ENIs could not assign the required resources, loop till the new ENIs can assign the required
 	// number of IPv4 resources
+	// Note: Windows don't support multi-ENI yet, so don't create new ENI even if primary ENI cannot assign required resources.
 	for len(assignedIPv4Resources) < required &&
-		len(e.attachedENIs) < eniLimit {
+		len(e.attachedENIs) < eniLimit &&
+		e.instance.Os() != config.OSWindows {
```

Comment:
What is this for? The controller doesn't provision IP resources for trunk/branch interface in Linux.

Reply:
Initially the existing code had implemented creating new ENI when there is not enough network interface to assign IPv4 resources on the primary ENI. However, Windows node doesn't support multi-ENI at the moment. Adding this check for `e.instance.Os() != config.OSWindows` here essentially skipped the following code but still kept the code there in case in the future when multi-ENI support for Windows is added, removing this check will re-enable the following code to create new ENI easily. Please let me know if you think it makes more sense to remove it entirely from the code,

Reply:
I think this equality check is misleading here. Commenting this code block with a TODO comment makes more sense to me.